go/client/cmd_simplefs_move.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package client
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+// CmdSimpleFSMove is the 'fs list' command.
+type CmdSimpleFSMove struct {
+	libkb.Contextified
+	src  []keybase1.Path
+	dest keybase1.Path
+}
+
+// NewCmdSimpleFSMove creates a new cli.Command.
+func NewCmdSimpleFSMove(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "mv",
+		ArgumentHelp: "<source> [source] <dest>",
+		Usage:        "copy one or more directory elements to dest",
+		Action: func(c *cli.Context) {
+			cl.ChooseCommand(&CmdSimpleFSMove{Contextified: libkb.NewContextified(g)}, "mv", c)
+		},
+	}
+}
+
+// Run runs the command in client/server mode.
+func (c *CmdSimpleFSMove) Run() error {
+	cli, err := GetSimpleFSClient(c.G())
+	if err != nil {
+		return err
+	}
+
+	ctx := context.TODO()
+
+	isDestDir, destPathString, err := getDirPathString(ctx, cli, c.dest)
+	if err != nil {
+		return err
+	}
+
+	for _, src := range c.src {
+		c.G().Log.Debug("SimpleFSMove %s -> %s, %v", pathToString(src), destPathString, isDestDir)
+
+		dest, err := makeDestPath(ctx, cli, src, c.dest, isDestDir, destPathString)
+		if err != nil {
+			return err
+		}
+		c.G().Log.Debug("SimpleFSMove %s -> %s", pathToString(src), pathToString(dest))
+
+		opid, err := cli.SimpleFSMakeOpid(ctx)
+		if err != nil {
+			return err
+		}
+		defer cli.SimpleFSClose(ctx, opid)
```

Comment:
What does `SimpleFSClose` do?  Does it **kill** an outstanding FS operation, or does it **wait** for it to complete?   My read is that it does a **kill**, which means that if you'll be accumulating a lot of parallel actions, and then **killing** them all as you leave the function.  The seems broken to me.